regolith/dates.py

```diff
@@ -168,62 +170,88 @@ def get_dates(thing):
     1 and 1, respectively
     '''
 
-    if thing.get("end_year") and not thing.get("begin_year"):
+    begin_year = "begin_year"
```

Comment:
this looks very ugly.  How about something like
```
datenames = ["day", "month", "year", "date"]
if date_field_prefix:
    datenames = [ "_".join(date_field_prefix datename) for datename in datenames]
```
then down below you can replace ``'day'`` with ``datenames[0]`` and so on.

The code for "begin_" and "end_" should not be touched....i.e., there is never ``"something_begin_day"``

A general tip, it is not considered good to concatanate strings using +.  Use f-strings or use join (btw, I may have the syntax wrong for the join, not sure if it takes an iterable)